### PR TITLE
added fix for #155

### PR DESCRIPTION
**Pull request by psnbaba**

### CODE REVIEW

File: pysph/base/particle_array.pyx [hunk 1]
```diff
@@ -569,6 +569,13 @@ cdef class ParticleArray:
         prop = ''
         for prop in particle_props:
             self._check_property(prop)
+            if prop in self.stride.keys():
+                stride = self.stride[prop]
+                prop_len = len(particle_props[prop])
+                base_len = len(particle_props['x'])
```
Comment (prabhuramachandran) [inline pysph/base/particle_array.pyx]: You can put this out of the loop and find the number of real particles as self.get_number_of_particles(), do you need the ghost particles also?

File: pysph/base/particle_array.pyx [hunk 2]
```diff
@@ -569,6 +569,13 @@ cdef class ParticleArray:
...
+                if not prop_len == base_len * stride:
+                    raise AttributeError, 'property %s length should be %d'%(prop,
```
Comment (prabhuramachandran) [inline pysph/base/particle_array.pyx]: Use `raise AttributeError(...)`

File: pysph/base/particle_array.pyx (hunk 1 above)
Comment (psnbaba) [reply pysph/base/particle_array.pyx]: I have put this inside loop as only the `prop` in `particle_props` need to be checked for the consistency. Poperties which are not present in `particle_props` will be initialized consistently with the stride value later in the function. No, I don't need ghost particles.

`self.get_number_of_particles()` will give me length of the current particle array. However, I want the length of `x` in the new particle_array which is passed.

Comment (prabhuramachandran) [reply pysph/base/particle_array.pyx]: You also do not need to do `if prop in self.stride.keys`, just `prop in self.stride` will do.

Comment (prabhuramachandran) [reply pysph/base/particle_array.pyx]: I mean you can get the base_len outside the loop.  Also what if the particle_props do not have 'x'?  What then?

Comment (psnbaba) [reply pysph/base/particle_array.pyx]: the new particles to be added should at least contain x, y and z. isn't it?
if not we should add that check too.

**Comment by psnbaba** [comment]
> There are still issues with this that I am not comfortable with, also the gpu cases need to be addressed.

actually I was getting garbage value due to this bug and took a lot of time to find the bug as I was new to pysph. So this need to be fixed. I try to write test cases. Also if @adityapb have time, I can use his help.

**Comment by adityapb** [comment]
> > There are still issues with this that I am not comfortable with, also the gpu cases need to be addressed.
> 
> actually, I was getting garbage value due to this bug and took a lot of time to find the bug as I was new to pysph. So this need to be fixed. I try to write test cases. Also if @adityapb have time, I can use his help for gpu cases.

@psnbaba I won't be on campus this week. We can meet on Monday and discuss this

File: pysph/base/particle_array.pyx [hunk 3]
```diff
@@ -569,6 +569,14 @@ cdef class ParticleArray:
         prop = ''
         for prop in particle_props:
             self._check_property(prop)
+            if prop in self.stride.keys():
+                stride = self.stride[prop]
+                prop_len = len(particle_props[prop])
+                base_len = self.get_number_of_particles()
+                print(stride, prop_len, base_len)
```
Comment (prabhuramachandran) [inline pysph/base/particle_array.pyx]: Remove the print statement.

File: pysph/base/particle_array.pyx [hunk 4]
```diff
@@ -569,6 +569,14 @@ cdef class ParticleArray:
...
+                if not prop_len == base_len * stride:
```
Comment (prabhuramachandran) [inline pysph/base/particle_array.pyx]: Write this as:
```
msg = 'Property %s length should be %d' % (prop, base_len * stride)
assert prop_len == base_len * stride, msg
```

**Comment by prabhuramachandran** [comment]
Also, the tests are failing on the GPU as in that case this check is not performed and an assertion error is not raised.

File: pysph/base/particle_array.pyx (hunk 3 above)
Comment (psnbaba) [reply pysph/base/particle_array.pyx]: Done

File: pysph/base/particle_array.pyx (hunk 4 above)
Comment (psnbaba) [reply pysph/base/particle_array.pyx]: Done

**Pull request merged**